_test/deploy-gatekeeper.sh

```diff
@@ -67,18 +67,36 @@ patch_namespaceselector_for_webhook() {
 generate_constraints() {
   echo "Creating ConstraintTemplates via konstraint..."
   konstraint doc -o POLICIES.md
+  konstraint create
 
-  # Ignores explained:
-  # - podman: the data is 'off-cluster' so cant be tested against gatekeeper
-  # - ocp/deprecated: tests cant be deployed to a 4.x cluster so cant be tested against gatekeeper
-  # - ocp/bestpractices/deploymentconfig-triggers-notset: OCP API-Server adds a default ConfigChange trigger by default so cant be tested against gatekeeper
-  # - ocp/bestpractices/rolebinding-roleref-apigroup-notset: OCP API-Server does not accept data matching this criteria but they are good for conftest when people are moving from 3.11 to 4.x
-  # - ocp/bestpractices/rolebinding-roleref-kind-notset: OCP API-Server does not accept data matching this criteria but they are good for conftest when people are moving from 3.11 to 4.x
-  # TODO LIST
-  # - ocp/bestpractices/container-resources-limits-memory-greater-than: TODO: https://github.com/redhat-cop/rego-policies/issues/76
-  # - ocp/bestpractices/container-resources-requests-memory-greater-than: TODO: https://github.com/redhat-cop/rego-policies/issues/76
-  # - combine: TODO: https://github.com/redhat-cop/rego-policies/issues/70
-  konstraint create --ignore "(podman|ocp\/deprecated|ocp\/bestpractices\/deploymentconfig-triggers-notset|ocp\/bestpractices\/rolebinding-roleref-apigroup-notset|ocp\/bestpractices\/rolebinding-roleref-kind-notset|ocp\/bestpractices\/container-resources-limits-memory-greater-than|ocp\/bestpractices\/container-resources-requests-memory-greater-than|combine)"
+  # shellcheck disable=SC2038
+  for file in $(find policy/* \( -name "template.yaml" -o -name "constraint.yaml" \) -type f | xargs); do
```

Comment:
I'd suggest creating constraint templates first and then apply constraints with sleep checks because in my case, `find` listed a constraint.yaml and tried to apply it without its template definition and got an error 🤦🏻‍♀️

Reply:
I had it like that previously, but still saw issues like that... the error checking needs improving probably, but what to - not sure. I'd rather not add `sleeps` because of how long do you sleep for? 5 seconds might be long enough on my cluster but you might need 1min....

Reply:
Ah sorry, don't know why I wrote sleep checks 🤦 Just wanted to point out that would be good if templates are ready before constraints :) yeah you're right, that'll be changed when prows comes in.

Reply:
So the `find` should create that order, as per:

```
MacBook-Pro:rego-policies garethhealy$ find policy/* \( -name "template.yaml" -o -name "constraint.yaml" \) -type f
policy/ocp/bestpractices/container-resources-requests-cpuunit-incorrect/template.yaml
policy/ocp/bestpractices/container-resources-requests-cpuunit-incorrect/constraint.yaml
policy/ocp/bestpractices/container-labelkey-inconsistent/template.yaml
policy/ocp/bestpractices/container-labelkey-inconsistent/constraint.yaml
policy/ocp/bestpractices/container-resources-memoryunit-incorrect/template.yaml
policy/ocp/bestpractices/container-resources-memoryunit-incorrect/constraint.yaml
policy/ocp/bestpractices/container-resources-limits-cpu-set/template.yaml
policy/ocp/bestpractices/container-resources-limits-cpu-set/constraint.yaml
policy/ocp/bestpractices/container-java-xmx-set/template.yaml
policy/ocp/bestpractices/container-java-xmx-set/constraint.yaml
policy/ocp/bestpractices/container-volumemount-missing/template.yaml
policy/ocp/bestpractices/container-volumemount-missing/constraint.yaml
policy/ocp/bestpractices/container-volumemount-inconsistent-path/template.yaml
policy/ocp/bestpractices/container-volumemount-inconsistent-path/constraint.yaml
policy/ocp/bestpractices/container-resources-limits-memory-notset/template.yaml
policy/ocp/bestpractices/container-resources-limits-memory-notset/constraint.yaml
policy/ocp/bestpractices/common-k8s-labels-notset/template.yaml
policy/ocp/bestpractices/common-k8s-labels-notset/constraint.yaml
policy/ocp/bestpractices/container-liveness-readinessprobe-equal/template.yaml
policy/ocp/bestpractices/container-liveness-readinessprobe-equal/constraint.yaml
policy/ocp/bestpractices/container-env-maxmemory-notset/template.yaml
policy/ocp/bestpractices/container-env-maxmemory-notset/constraint.yaml
policy/ocp/bestpractices/container-secret-mounted-envs/template.yaml
policy/ocp/bestpractices/container-secret-mounted-envs/constraint.yaml
policy/ocp/bestpractices/container-image-latest/template.yaml
policy/ocp/bestpractices/container-image-latest/constraint.yaml
policy/ocp/bestpractices/container-livenessprobe-notset/template.yaml
policy/ocp/bestpractices/container-livenessprobe-notset/constraint.yaml
policy/ocp/bestpractices/pod-replicas-not-odd/template.yaml
policy/ocp/bestpractices/pod-replicas-not-odd/constraint.yaml
policy/ocp/bestpractices/container-readinessprobe-notset/template.yaml
policy/ocp/bestpractices/container-readinessprobe-notset/constraint.yaml
policy/ocp/bestpractices/pod-hostnetwork/template.yaml
policy/ocp/bestpractices/pod-hostnetwork/constraint.yaml
policy/ocp/bestpractices/pod-replicas-below-one/template.yaml
policy/ocp/bestpractices/pod-replicas-below-one/constraint.yaml
policy/ocp/requiresinventory/deployment-has-matching-service/template.yaml
policy/ocp/requiresinventory/deployment-has-matching-service/constraint.yaml
policy/ocp/requiresinventory/deployment-has-matching-pvc/template.yaml
policy/ocp/requiresinventory/deployment-has-matching-pvc/constraint.yaml
policy/ocp/requiresinventory/service-has-matching-servicemonitor/template.yaml
policy/ocp/requiresinventory/service-has-matching-servicemonitor/constraint.yaml
policy/ocp/requiresinventory/deployment-has-matching-serviceaccount/template.yaml
policy/ocp/requiresinventory/deployment-has-matching-serviceaccount/constraint.yaml
policy/ocp/requiresinventory/deployment-has-matching-poddisruptionbudget/template.yaml
policy/ocp/requiresinventory/deployment-has-matching-poddisruptionbudget/constraint.yaml
```

whats probably happened, is gatekeeper has said its accepted the template, the checks have said there are no errors but actually, gatekeeper wasn't fully running yet so didn't accept the template. I've seen that happen but not sure how to fix it.

we need a check that says `100%, gatekeeper is running`

Reply:
Oh, I see! We'll think about on it then 🤓🤓